src/solver/bounded.rs

```diff
@@ -0,0 +1,1429 @@
+                    ("1", set![]),
+                    ("2", set![]),
+                    ("3", set![Element::new(vec![])]),
+                    ("4", set![])
+                ]),
+                state([
+                    ("1", set![]),
+                    ("2", set![]),
+                    ("3", set![]),
+                    ("4", set![Element::new(vec![])])
+                ]),
+            ])
+        );
+        assert_eq!(result5, result4);
+    }
+
+    #[test]
+    fn interpreter_cycle_vals() {
+        let program = Program {
+            inits: set![state([("s", set![Element::new(vec![0])])])],
+            trs: set![
+                Transition::new(
+                    set![Guard::includes("s", Element::new(vec![0]))],
+                    set![
+                        Update::remove("s", Element::new(vec![0])),
+                        Update::insert("s", Element::new(vec![1]))
+                    ],
+                ),
+                Transition::new(
+                    set![Guard::includes("s", Element::new(vec![1]))],
+                    set![
+                        Update::remove("s", Element::new(vec![1])),
+                        Update::insert("s", Element::new(vec![2]))
+                    ],
+                ),
+                Transition::new(
+                    set![Guard::includes("s", Element::new(vec![2]))],
+                    set![
+                        Update::remove("s", Element::new(vec![2])),
+                        Update::insert("s", Element::new(vec![3]))
+                    ],
+                ),
+                Transition::new(
+                    set![Guard::includes("s", Element::new(vec![3]))],
+                    set![
+                        Update::remove("s", Element::new(vec![3])),
+                        Update::insert("s", Element::new(vec![0]))
+                    ],
+                )
+            ],
+            safes: set![set![Guard::excludes("s", Element::new(vec![3]))]],
+        };
+        let result0 = interpret(&program, 0);
+        let result1 = interpret(&program, 1);
+        let result2 = interpret(&program, 2);
+        let result3 = interpret(&program, 3);
+        let result4 = interpret(&program, 4);
+        assert_eq!(result0, InterpreterResult::Unknown);
+        assert_eq!(result1, InterpreterResult::Unknown);
+        assert_eq!(result2, InterpreterResult::Unknown);
+        assert_eq!(
+            result3,
+            InterpreterResult::Counterexample(vector![
+                state([("s", set![Element::new(vec![0])])]),
+                state([("s", set![Element::new(vec![1])])]),
+                state([("s", set![Element::new(vec![2])])]),
+                state([("s", set![Element::new(vec![3])])]),
+            ])
+        );
+        assert_eq!(result4, result3);
+    }
+
+    fn handle_error<A, B: std::fmt::Display>(result: Result<A, B>) -> A {
+        match result {
+            Ok(a) => a,
+            Err(b) => panic!("{}", b),
```

Comment:
Why this helper function? You could instead use `Result::expect` as a method call or perhaps even have your tests produce a Result and just use `?`.

Reply:
It didn't replace `.expect("....")`, it was replacing `.unwrap_or_else(|e| panic!("{}", e))`. I forgot that tests could return things though, `?` looks nicer.